tests/acceptance.rs

```diff
@@ -0,0 +1,161 @@
+use assert_cmd::Command;
+use serde_json::Value;
+use serde_json_path::JsonPath;
+use std::env::current_dir;
+
+// Acceptance tests for zizmor, on top of Json output
+// For now we don't cover tests that depends on Github API under the hood
+
+fn zizmor() -> Command {
+    Command::cargo_bin("zizmor").expect("Cannot create executable command")
+}
+
+fn workflow_under_test(name: &str) -> String {
+    let current_dir = current_dir().expect("Cannot figure out current directory");
+
+    let file_path = current_dir.join("tests").join("test-data").join(name);
+
+    file_path
+        .to_str()
+        .expect("Cannot create string reference for file path")
+        .to_string()
+}
+
+fn assert_value_match(json: &Value, path_pattern: &str, value: &str) {
+    let json_path = JsonPath::parse(path_pattern).expect("Cannot evaluate json path");
+    let queried = json_path
+        .query(json)
+        .exactly_one()
+        .expect("Cannot query json path");
+
+    // Don't bother about surrounding formatting
+    assert!(queried.to_string().contains(value));
+}
+
+#[test]
+fn audit_artipacked() -> anyhow::Result<()> {
+    let auditable = workflow_under_test("artipacked.yml");
+    let cli_args = ["--format", "json", &auditable];
+
+    let execution = zizmor().args(cli_args).unwrap();
```

Comment:
Quick note : this `unwrap()` is not the same one from `Result` or `Option`. 

`args()` returns a `Command` type, hence there is no way to propagate an error here

Reply:
Makes sense, thanks for explaining. Also TIL that `#[test]` functions can return result types!